client/me/concierge/calendar-card.js

```diff
@@ -39,9 +39,37 @@ class CalendarCard extends Component {
 		timezone: PropTypes.string.isRequired,
 	};
 
-	state = {
-		selectedTime: this.props.times[ 0 ],
-	};
+	constructor( props ) {
+		super( props );
+
+		// To find the best default time for the time picker, we're going to pick the time that's
+		// closest to the current time of day. To do this first we find out how many seconds it's
+		// been since midnight on the current real world day...
+		const millisecondsSinceMidnight = moment().diff( this.withTimezone().startOf( 'day' ) );
+
+		// Then we'll use that to find the same time of day on the Card's given date. This will be the
+		// target timestamp we're trying to get as close to as possible.
+		const targetTimestamp = this.withTimezone( this.props.date )
+			.startOf( 'day' )
+			.add( millisecondsSinceMidnight );
+
+		// Default to the first timestamp and calculate how many seconds it's offset from the target
+		let closestTimestamp = this.props.times[ 0 ];
```

Comment:
nitpick: it would crash if somehow we pass in an empty array as `times`.

Reply:
Good observation! Empty arrays are regularly passed in when there's no times available on a given day, so this is common. I'm going to wrap this in a conditional so the code is clearer, but as a fun FYI this actually works fine with an empty array.

If you address non-existant indexes of an array the result is `undefined`, so this will just set `closestTimestamp` to `undefined`. Looking further down, `let closestTimeOffset = Math.abs( closestTimestamp - targetTimestamp );` will end up as `NaN` because you've got `undefined - <some number>`. So that's not ideal, but since `times` is empty this value will never be used anyways.

In the end you'll have `this.state.selectedTime === undefined` which is fine. But since it's not clear that this _should_ work, I'm going to wrap it in a conditional :)

Reply:
#js-magic